server/src/main/java/org/elasticsearch/repositories/blobstore/ChecksumBlobStoreFormat.java

```diff
@@ -128,7 +129,8 @@ public T deserialize(String blobName,
                 }
             } else {
                 StreamInput streamInput = CompressorFactory.COMPRESSOR.isCompressed(slice)
-                    ? CompressorFactory.COMPRESSOR.uncompress(slice).streamInput()
+                    // Ensure that 5.10+ uses threadLocalInputStream, similar to 5.9
+                    ? new InputStreamStreamInput(CompressorFactory.COMPRESSOR.threadLocalInputStream(slice.streamInput()))
```

Comment:
This is inspired by `PublicationTransportHandler.handleIncomingPublishRequest` and heap dump was showing that
NamedWriteableAwareStreamInput's delegate is `InputStreamStreamInput` as well.

Difference is only how we create this InputStreamStreamInput: 

In this PR it's from a local `threadLocalInputStream` and was something like 
`compressor.uncompress` ->  use `inappropriate inflater` -> `ref.copyBytes` -> `BytesArray.streamInput` ->  `StreamInput.wrap` -> `InputStreamStreamInput`